scripts/speaker_tasks/scp_to_manifest.py

```diff
@@ -146,52 +173,28 @@ def get_labels(lines):
     return labels
 
 
-def write_audio_file(line):
-    filename = line['audio_filepath']
-    label = line['label']
-    offset = line['offset']
-    duration = line['duration']
-    basename = os.path.basename(filename).replace('.wav', '')
-    to_path = os.path.join(CWD, 'chunks', label)
-    os.makedirs(to_path, exist_ok=True)
-    to_path = os.path.join(to_path, basename)
-    final_string = '_' + str(offset) + '_' + str(duration)
-    final_string = final_string.replace('.', '-')
-    samples, sr = sf.read(filename)
-    start = int(float(offset * sr))
-    end = start + int(float(duration * sr))
-    chunk = samples[start:end]
-    to_file = to_path + final_string + '.wav'
-    sf.write(to_file, chunk, sr)
-
-    line['offset'] = 0
-    line['audio_filepath'] = to_file
-    return line
-
-
-def main(scp, manifest, id, out, split=False, create_chunks=False, write_chunks=False, min_count=10, workers=4):
+def main(scp, manifest, id, out, split=False, create_chunks=False, min_count=10):
     if os.path.exists(out):
         os.remove(out)
     if scp:
         lines = read_file(scp_file=scp, id=id)
+        lines = process_map(get_duration, lines, chunksize=100)
+        out_file = os.path.splitext(scp)[0] + '_manifest.json'
+        write_file(out_file, lines, range(len(lines)))
     else:
         lines = read_manifest(manifest)
 
     lines = process_map(get_duration, lines, chunksize=100)
 
     if create_chunks:
-        print("creating chunk")
+        print("creating and writing chunks to {}".format(CWD))
```

Comment:
https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md
10. F-strings are preferred to formatted strings.

I recently figured that you have already used lots of str.format() method here and there.  Following the NeMo contributing guide, I suggest you to use f-string. Which makes -> `print(f"creating and writing chunks to {CWD}"`

str.format() appeared in python 2.6.

Reply:
Thanks tj, updated the code to use f-string.